spyder/widgets/pathmanager.py

```diff
@@ -21,285 +23,431 @@
+        """
+        if directory is None:
+            self.redirect_stdio.emit(False)
+            directory = getexistingdirectory(self, _("Select directory"),
+                                             self.last_path)
+            self.redirect_stdio.emit(True)
 
-    def update_list(self):
-        """Update path list"""
-        self.listwidget.clear()
-        for name in self.pathlist+self.ro_pathlist:
-            item = QListWidgetItem(name)
-            item.setIcon(ima.icon('DirClosedIcon'))
-            if name in self.ro_pathlist:
-                item.setFlags(Qt.NoItemFlags | Qt.ItemIsUserCheckable)
-                item.setCheckState(Qt.Checked)
-            elif name in self.not_active_pathlist:
-                item.setFlags(item.flags() | Qt.ItemIsUserCheckable)
-                item.setCheckState(Qt.Unchecked)
+        if PY2:
+            is_unicode = False
+            try:
+                directory.decode('ascii')
+            except (UnicodeEncodeError, UnicodeDecodeError):
+                is_unicode = True
+
+            if is_unicode:
+                QMessageBox.warning(
+                    self,
+                    _("Add path"),
+                    _("You are using Python 2 and the path selected has "
+                      "Unicode characters."
+                      "<br> "
+                      "The new path will not be added."),
+                    QMessageBox.Ok)
+                return
+
+        directory = osp.abspath(directory)
+        self.last_path = directory
+
+        if directory in self.get_path_dict():
+            item = self.listwidget.findItems(directory, Qt.MatchExactly)[0]
+            item.setCheckState(Qt.Checked)
+            answer = QMessageBox.question(
+                self,
+                _("Add path"),
+                _("This directory is already included in the list."
+                  "<br> "
+                  "Do you want to move it to the top of the list?"),
+                QMessageBox.Yes | QMessageBox.No)
+
+            if answer == QMessageBox.Yes:
+                item = self.listwidget.takeItem(self.listwidget.row(item))
+                self.listwidget.insertItem(0, item)
+                self.listwidget.setCurrentRow(0)
+        else:
+            if self.check_path(directory):
+                item = self._create_item(directory)
+                self.listwidget.insertItem(0, item)
+                self.listwidget.setCurrentRow(0)
             else:
-                item.setFlags(item.flags() | Qt.ItemIsUserCheckable)
-                item.setCheckState(Qt.Checked)
-            self.listwidget.addItem(item)
+                answer = QMessageBox.warning(
+                    self,
+                    _("Add path"),
+                    _("This directory cannot be added to the path!"
+                      "<br><br>"
+                      "If you want to set a different Python interpreter, "
+                      "select one in the preferences."),

```

Comment:
Please change this to

    please go to <tt>Preferences > Main interpreter</tt>